libraries/rush-lib/src/api/Subspace.ts

```diff
@@ -56,6 +61,23 @@ export class Subspace {
     return this._projects;
   }
 
+  public getPnpmOptions(): PnpmOptionsConfiguration | undefined {
```

Comment:
Since this deals with the filesystem, this function should be async.

Reply:
That should probably happen anyway. How much farther up the chain would that have to go?

Reply:
> That should probably happen anyway. How much farther up the chain would that have to go?

@iclanton @william2958  The `Subspace` class is a public API that was introduced as a refactoring of existing Rush code which was originally synchronous.    Keep in mind that: 

🚨**Synchronous code cannot call asynchronous APIs**. 
👍Asynchronous code CAN call synchronous APIs.

Thus, if we want to apply `async` across all public APIs involved with the filesystem, then we must also apply `async` to any APIs that /might/ be involved in calling such `async` APIs.

Thus for public APIs, it might be better to do one big backwards-compatibility-breaking PR that `async`-ifies all the APIs.  Whereas for internal-only code, it makes sense to incrementally fix up functions as we work on them.

Reply:
@octogonz so should it be left as is? I looked at it and there aren't too many changes required to make this change, just adding await keywords in one line in RushInstallManager and WorkspaceInstallManager (other than making the InstallHelper.generateCommonPackageJson async)

Reply:
If it's a new API and you can make it `async` easily, sure, go for it.